src/Microsoft.OpenApi/Models/OpenApiSchema.cs

```diff
@@ -715,7 +715,7 @@ private void SerializeAsV2(
             // readOnly
             // In V2 schema if a property is part of required properties of parent schema,
             // it cannot be marked as readonly.
-            if (!parentRequiredProperties.Contains(propertyName))
+            if (propertyName is not null && !parentRequiredProperties.Contains(propertyName))
```

Comment:
string is null or empty instead?

Reply:
Unfortunately `string.IsNullOrEmpty` still treats a nullableString as string?, resulting in a nullability warning. The compiler still doesn't recognize that the nullable string must be non-null if the condition passes hence the explicit null check

Reply:
same rationale as previous comment